### PR TITLE
Properly install public headers

### PR DESCRIPTION
**Pull request by jupp0r**
This change exposes public headers. In the future the amout of exposed headers could be reduced my using private implementation techniques.

### CODE REVIEW

**Comment by gjasny** [comment]
Why don't you use `prometheus-cpp` as header prefix?

**Comment by jupp0r** [comment]
I like the namespace matching the include directory.

File: include/prometheus/handler.h [hunk 1]
```diff
@@ -3,10 +3,10 @@
 #include <memory>
 #include <vector>
 
-#include "CivetServer.h"
+#include <prometheus/collectable.h>
```
Comment (gjasny) [inline include/prometheus/handler.h]: Could you include your own headers with either `"prometheus/collectable.h"` or just a plain `"collectable.h"`.

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -57,6 +45,7 @@ add_library(prometheus-cpp
 
 # TODO(gj) make all PRIVATE
 target_link_libraries(prometheus-cpp PUBLIC ${PROTOBUF_LIBRARIES})
+target_include_directories(prometheus-cpp PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include> $<INSTALL_INTERFACE:include>)
```
Comment (gjasny) [inline lib/CMakeLists.txt]: There is no need for the `$<INSTALL_INTERFACE>` genex here because the `INCLUDES DESTINATION ...` below does the same thing.

**Pull request merged**